core/db/migrate/20210407200948_create_spree_menus.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreeMenus < ActiveRecord::Migration[6.0]
+  def change
+    create_table :spree_menus do |t|
+      t.column :name, :string
+      t.column :unique_code, :string, null: false
+
+      t.timestamps
+    end
+
+    add_index :spree_menus, :name
+    add_index :spree_menus, :unique_code, unique: true
```

Comment:
```suggestion
    add_index :spree_menus, :unique_code
```

let's have only rails unique validation here